posts/copilot-workspace/index.qmd

```diff
@@ -0,0 +1,102 @@
+    orcid: "0000-0001-8814-9421"
+date: "2024-06-24"
+categories: [AI, GitHub, Copilot Workspace, R, R package, software development]
+bibliography: index.bib
+format:
+  html: 
+    toc: true
+---
+
+
+GitHub recently previewed ‘Copilot Workspace’, which aims to use generative AI to assist software developers. Rather than just giving in-line suggestions, as GitHub copilot does, workspace allows users to map out and execute entire projects. We got early preview access to the tool, so decided to see how well it performed for our needs in Epiverse.
+
+In the Epiverse-TRACE initiative we aim to develop an ecosystem of epidemic analytics tools as [Digital Public Goods](https://digitalpublicgoods.net/) in the form of R packages. We aim to [follow best-practices](https://epiverse-trace.github.io/blueprints/) given that our tools will be used to inform  public health decisions. For efficiency, we use a range of development tools, including integrated development environments (IDE), such as RStudio and Visual Studio Code, other R packages to help check and document our code, for example {lintr} and {roxygen2}, and version control and code sharing tools, git and GitHub.
+
+Given the rapid development and increasing availability to artificial intelligence (AI) models, specifically large language models, offer the potential to be another development tool to add to the arsenal of researchers and software engineers to enhance productivity and improve the reliability of the code we write and others depend upon [@vanHoek2024]. Large language models like ChatGPT from OpenAI, Gemini from Google, Claude from Anthropic, as well as many others and new ones frequently appearing, have presented opportunities to quickly generate material -- text, code and images -- from text prompts.
+
+We decided to explore the suitability of one such, GitHub Copilot integrated with GitHub Workspaces, for some typical use cases encountered by our research software engineering team. We do note that many other such tools are available, and we chose Copilot in Workspaces as our code is hosted on GitHub, making for easy evaluation of the specific contribution of the AI tool. However, some of our broad conclusions are based on using other tools for similar tasks.
+
+To test this out, a small group of Epiverse-TRACE professors and research software engineers explored its functionality. The aim of this blog post is to summarise our experience with Copilot and Workspaces, and share our thoughts more generally about how the current feature set of AI-powered development tools could help with our daily development tasks, and with research software development at large. We evaluated Copilot workspace across three tasks of varying difficulty and requirements:
+
+* Experiment 1: Exporting an R function by updating the documentation
+
+* Experiment 2: Adding a basic Susceptible-Infected-Recovered (SIR) model to an existing R package that contains other epidemiological models
```

Comment:
Given the audience could be anyone, we may want to reduce distractions with niche words.

```suggestion
* Experiment 2: Adding a basic epidemic model to an existing R package that contains other epidemiological models
```